src/libertem/web/shutdown.py

```diff
@@ -0,0 +1,18 @@
+import tornado
+import logging
+
+from .state import SharedState
+
+log = logging.getLogger(__name__)
+
+
+class ShutdownHandler(tornado.web.RequestHandler):
+    def initialize(self, state: SharedState, event_registry):
+        self.state = state
+        self.event_registry = event_registry
+
+    async def get(self):
```

Comment:
This should possibly be a non-`GET` method, as `GET` is defined as a "safe method", which [SHOULD NOT have the significance of taking an action other than retrieval.](https://tools.ietf.org/html/rfc2616#section-9) (excuse the shouting, that's part of how RFC specifications are written...) `POST` or `DELETE` may be more appropriate. Also, possibly write a response (and wait until it is completely sent) before shutting down the I/O loop. The [flush method](https://www.tornadoweb.org/en/stable/web.html#tornado.web.RequestHandler.flush) may be of help here.

Rest of the `RequestHandler` looks good.

Reply:
>
> This should possibly be a non-`GET` method, as `GET` is defined as a "safe method"

Thx for the information.

> Also, possibly write a response (and wait until it is completely sent) before shutting down the I/O loop.

What is the purpose of sending the response? (Is that to ensure we are inside request handler 
and about to shutdown)

Reply:
> What is the purpose of sending the response?

This is to inform the client that the shutdown request was received and will be processed. If you don't send a response, the client will never know - maybe the network connection just dropped before receiving the request, and the server is still running?

Reply:
Under what circumstance we use `RequestHandler.finish()`

Reply:
After sending the response, to make sure we wait for the response to be sent before the event loop is stopped. Otherwise we may only send a part of the reponse, or nothing at all. [RequestHandler.write](https://www.tornadoweb.org/en/stable/web.html#tornado.web.RequestHandler.write) only writes into an output buffer, so it is not guaranteed that the response ends up on the network after only calling `write`.